src/iscn/parsing.ts

```diff
@@ -0,0 +1,97 @@
+import BigNumber from 'bignumber.js';
+import { getLikeWalletAddress, isValidAddress, changeAddressPrefix } from './addressParsing';
+import { getLikeWalletByLikerId } from './likerIdsAddresses';
+import { ISCNRecordData } from '../types';
+
+async function getLikeWalletFromId(
+  id: string,
+  { LIKE_CO_API_ROOT = 'https://api.like.co' }: { LIKE_CO_API_ROOT?: string } = {},
+)
+: Promise < { likeWallet: string | void | null } > {
+  let likeWallet: string | void | null = null;
+  let likerId: string | void | null = null;
+  const res = id.match(/^https:\/\/like\.co\/([a-z0-9_-]{6,20})/);
+  if (res) {
+    [, likerId] = res;
+    likeWallet = await getLikeWalletByLikerId(likerId, { LIKE_CO_API_ROOT });
+  } else {
+    likeWallet = getLikeWalletAddress(id);
+  }
+  return { likeWallet };
+}
+
+export async function getStakeholderMapFromParsedIscnData(
+  iscnData: ISCNRecordData,
+  { totalLIKE = 1, defaultWallet }: { totalLIKE?: number, defaultWallet?: string } = {},
+)
+: Promise < Map < string, { LIKE: number } > > {
+  const LIKEMap = new Map();
+  const { stakeholders } = iscnData;
+  if (stakeholders?.length) {
+    const likeWallets = stakeholders.map((stakeholder:any) => {
+      const id: string = stakeholder.entity['@id'];
+      if (isValidAddress(id)) {
+        return changeAddressPrefix(id, 'like');
+      }
+      return null;
+    });
+    const weightMap = new Map();
+    likeWallets.forEach((element: any, i) => {
```

Comment:
Suggest using `likeWallet` instead of `element`, this provides more context for reader